crates/router/src/core/payments/flows/setup_mandate_flow.rs

```diff
@@ -109,7 +109,8 @@ impl Feature<api::SetupMandate, types::SetupMandateRequestData> for types::Setup
                 key_store,
                 is_mandate,
             ))
-            .await?;
+            .await?
+            .0;
```

Comment:
We need to send back the payment_method_id and payment_method_status in Setup Mandate flow as well. Please ensure you set it accordingly.